src/app/api/v1/slack/rotation/on-holiday-post/route.ts

```diff
@@ -0,0 +1,57 @@
+import { PostgresClient } from 'utils/database';
+import { NextRequest, NextResponse } from 'next/server';
+import {
+    getSlackMessage,
+    SlackResponseType,
+    sanitizeSlackText,
+    parsePayloadFromRequest,
+} from 'utils/slack';
+
+export async function POST(req: NextRequest) {
+    try {
+        const parsedPayload = await parsePayloadFromRequest(req);
+        const { text, team_domain } = parsedPayload;
+
+        // Extract and sanitize Slack ID, convert to uppercase, else it breaks
+        const slackIdMatch = text.match(/<@([A-Z0-9]+)>/);
+        const slackId = slackIdMatch ? sanitizeSlackText(slackIdMatch[1]).toUpperCase() : null;
+
+        const rotationName = sanitizeSlackText(text.replace(/<@[^>]+>/g, '').trim());
+
+        if (!rotationName) {
+            return NextResponse.json({ error: 'rotationName is required' }, { status: 400 });
+        }
+
+        if (!slackId) {
+            return NextResponse.json({ error: 'Slack user ID is required' }, { status: 400 });
+        }
+
+        const organizationName = sanitizeSlackText(team_domain);
+
+        const DbClient = new PostgresClient(organizationName, rotationName);
+
+        const user = await DbClient.toggleHolidayStatus(slackId);
+
+        if (!user) {
+            return NextResponse.json(
+                getSlackMessage(SlackResponseType.Ephemeral, 'User not found or invalid.'),
+                { status: 404 }
+            );
+        }
```

Comment:
I guess that the error messages thing can be abstracted, if any of the user, slackId, or organization don't exists.